common/src/main/java/com/wynntils/screens/maps/widgets/IconFilterWidget.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.screens.maps.widgets;
+
+import com.google.common.collect.Lists;
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.screens.maps.IconFilterScreen;
+import com.wynntils.utils.EnumUtils;
+import com.wynntils.utils.colors.CommonColors;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.render.RenderUtils;
+import com.wynntils.utils.render.Texture;
+import java.util.List;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.client.gui.components.AbstractWidget;
+import net.minecraft.client.gui.narration.NarrationElementOutput;
+import net.minecraft.network.chat.Component;
+
+public class IconFilterWidget extends AbstractWidget {
+    private final boolean included;
+    private final float iconRenderX;
+    private final float iconRenderY;
+    private final List<Component> tooltip;
+    private final IconFilterScreen filterScreen;
+    private final Texture icon;
+
+    public IconFilterWidget(
+            int x, int y, int width, int height, Texture icon, IconFilterScreen filterScreen, boolean included) {
+        super(x, y, width, height, Component.literal("Icon Filter Widget"));
+        this.icon = icon;
+        this.filterScreen = filterScreen;
+        this.included = included;
+
+        // Calculate x/y position of the icon to keep it centered
+        iconRenderX = (x + width / 2f) - icon.width() / 2f;
+        iconRenderY = (y + height / 2f) - icon.height() / 2f;
```

Comment:
I think this should be done at render time to keep these things together.

Reply:
It only needs to be calculated once, same with the tooltip so no need to do it at render time